### ISSUE TITLE
Mammal recording form is missing location name

### ISSUE THREAD

**Issue by kitenetter** [BiologicalRecordsCentre/iRecord#1296]
I have been reminded that we have a mammal recording form hosted on the BRC website:
https://www.brc.ac.uk/mammals

This is the main recording form that the Mammal Society links to from their website:
https://www.mammal.org.uk/science-research/record-submission/

The issue that has been reported is that there used to be a field for location name, and that this has recently disappeared.

Also, data entered via this form is not appearing on iRecord.

The form feeds in to survey ID 35, "Mammal Atlas".

It looks like a new website has been set up, called "BRC website's forms" (ID = 145), but there is no sharing agreement.

Actions:
- Reinstate the location name attribute
- It would make more sense to have this form hosted on iRecord, is there any reason not to do that? Can we move the survey so that it is linked directly to iRecord?
- Which website was this survey originally linked to, and was there a website agreement in place?

**Comment by BirenRathod** [comment]
@kitenetter I can move this form on to iRecord but then the URL should change on to their website too.

Are you going to do that?

**Comment by kitenetter** [comment]
@BirenRathod I can liaise with Mammal Soc to change the link on their website.

Before we do anything else, am I right in thinking that the "Mammal Atlas" survey was originally linked to website ID 64 "Mammal Society CATlas"? If so, can we move the survey back into that website? If we do that I'm hoping that it will result in the records reappearing on iRecord and will also reinstate any sensitive species settings.

**Comment by BirenRathod** [comment]
@kitenetter I just checked in the database it was link to 'Mammal society' which I linked it back to it.

**Comment by kitenetter** [comment]
That should be fine, thanks. In which case I think we can go ahead and set up the recording form in iRecord. Suggest that we use this alias:
/enter-mammal-records

**Comment by BirenRathod** [comment]
@BirenRathod Ok, I will do that and let you know soon.

**Comment by BirenRathod** [comment]
@kitenetter Although, I reverted it back the survey to 'Mammal society', it is not showing in irecord.

@johnvanbreda I don't remember at present, but if I add 'mammal society' website in 'website agreements', would that display the survey in iRecord?

**Comment by BirenRathod** [comment]
Or shall I just change the website id from the database and set to irecord one?

**Comment by kitenetter** [comment]
From my perspective I just want to get back to where we were before the change! If the survey was previously linked to the Mammal Society I would prefer to keep it the same. Once that is done I would expect the records to start appearing on iRecord.

The Mammal Society website already has a sharing agreement in place.

**Comment by BirenRathod** [comment]
@kitenetter Is that ok if I change website id to irecord one instead of 'Mammal society'?

**Comment by BirenRathod** [comment]
@kitenetter Not sure that sharing agreement works same way for submitting record.

**Comment by kitenetter** [comment]
Why do we need to change it? If we do it will mean that we'l have to find a way of sorting out the sensitive species issues (see #1007), and it would also mean that the source given in the record downloads that the Mammal Society verifiers use will change.

Is there any reason we can't stay with the Mammal Society as the website?

**Comment by BirenRathod** [comment]
@kitenetter because survey is not available to use in irecord.

**Comment by BirenRathod** [comment]
@kitenetter just in case, if you are not clear, it is not showing in the survey list. see below screenshot.

![image](https://user-images.githubusercontent.com/5941255/169266220-9ccd3b57-7f9a-43b0-a817-14812f65efd0.png)

**Comment by kitenetter** [comment]
Ah, sorry, I should have realised that. In which case things are more complicated than I had realised.

@johnvanbreda for the sensitive species settings, are we able to apply them to a particular survey, or can they only be aplied to an entire website?

**Comment by johnvanbreda** [comment]
@kitenetter the workflow sensitive species handling is filtered on a website, not a survey.

As far as I can see, the brc.ac.uk/mammals form is linked to survey id 35 which is part of website 14 (Mammal Society). But the iform configuration on brc.ac.uk has been changed to point it to website ID 145 (BRC website's forms). Because the join table (sample_attributes_websites) that links survey 35 to it's custom attributes also specifies the website ID, they are not being found because it is looking for website ID 145 which is wrong. You can't just change the website ID on a Drupal site and expect all the configured forms to continue working against their surveys without fixing all the links in the database.

**Comment by BirenRathod** [comment]
@johnvanbreda That iform configuration, it is old one. it is not any longer there.
Where did you find it?

**Comment by johnvanbreda** [comment]
I right clicked on the form to view source so I can see it is survey ID 35, website ID 145. Then I looked in the database to compare and I can see it is wrong.

**Comment by BirenRathod** [comment]
@johnvanbreda  that changed to website id 14, so ignore that. 

Can you just check what Martin has asked above? we want to move this form on to irecord website.

**Comment by johnvanbreda** [comment]
I've answered the question about sensitivity control above.

You also asked:

> I don't remember at present, but if I add 'mammal society' website in 'website agreements', would that display the survey in iRecord?

Yes, that's correct. But, you may need to correct the Indicia cache first.

**Comment by BirenRathod** [comment]
ok, thanks @johnvanbreda. Sorry bit confused with other bit. Anyway I will do that and see if that works.

**Comment by kitenetter** [comment]
I'm now very confused.

Having the mammal form on the iRecord website is a nice-to-have, but is not essential.
Having the sensitive species workflow in place definitely is essential, so if we can't do that on iRecord we will have to leave the form on the BRC website.

But are we saying that it is possible to link survey ID 35 back to website ID 14 (which is where it started life), and then to display a recording form in iRecord that retains the link to website ID 14?

Let me know if we need a quick zoom, I'm just trying to get back to how things were before the change to the BRC website but I don't know how to get there!

**Comment by johnvanbreda** [comment]
It is possible to have a different website authentication on different forms on the brc.ac.uk website, but it is not something I would normally advise (I've only done this in one other situation). So if brc.ac.uk is going to act as a container for several recording forms, then they should probably all have website ID 145. This does mean they will all have the same sharing settings though - is that OK?

Therefore I think that the following needs to be done (probably using pgAdmin):
1. Change the website ID in survey 35 to 145 instead of 14.
2. For all the sample_attributes_websites and occurrence_attributes_websites records that link the custom attributes to the survey, if the restrict_to_survey_id is 35, change the website ID to 145.
3. Use the work_queue to re-build the cache data for all records in survey_id 35.

**Comment by BirenRathod** [comment]
@johnvanbreda brc.ac.uk will act as container for several recording forms in the future. Not sure which one those. I only added one and 'mammal atlas' survey added too but Martin can't access the records on iRecord.
Also, I will do what you suggested above. 

@kitenetter , is that ok?

**Comment by kitenetter** [comment]
@BirenRathod I think that is okay.

But I think this also means that:
- We need to include website 145 in the sensitive species workflow alongside website ID 14 - see #1007 
- We need to set up a website agreement for website 145 to share records with iRecord for reporting + verification + dataflow
- We need to add website 145 to the list of websites that get included in the ES dataset

However, that will mean that any other survey that is linked to website 145 will also be shared for reporting + verification + dataflow. Is that okay? What other surveys are in place or planned?

**Comment by BirenRathod** [comment]
@kitenetter At present, it has only got one survey which is Treelogs. 

Regarding you other queries, I would suggest I will change it to irecord one which is 23 instead of 145. And I will setup survey on irecord website which was original query. 

Shall I go ahead and do it?

**Comment by kitenetter** [comment]
No, sorry! We can't put it into website ID 23 because that makes it impossible to control the sensitive species settings.

Not sure what to suggest.

**Comment by BirenRathod** [comment]
@kitenetter In that case, I will use it 145 and can you look after other bit as I'm not able to do that.

**Comment by johnvanbreda** [comment]
If it is on website 145, then all the surveys on that website will share the same sensitivity workflow. This may not matter though as the other survey is not for mammals?

**Comment by BirenRathod** [comment]
@kitenetter I have re-updated the website id to 145 for survey 35. 

@johnvanbreda Why do I need to do  2 & 3 steps?

**Comment by BirenRathod** [comment]
@kitenetter I also added the location name.

**Comment by johnvanbreda** [comment]
Step 2 was to make sure the attributes link to the correct website.
Step 3 was to make sure the cache data is updated for the old records.

**Comment by BirenRathod** [comment]
@johnvanbreda I think all attributes are linked if I'm not wrong as people are already submitting record. so should be fine I think.

**Comment by BirenRathod** [comment]
@johnvanbreda No problem. Thanks.

@kitenetter can you see all records submitted using website_id 145? Or do you need to setup the agreement to see those record on iRecord. There are 116 records.

**Comment by kitenetter** [comment]
I can't see anything at the moment. I'll set up the agreement.

What is the "Tree logs" survey, who is running that?

**Comment by BirenRathod** [comment]
@kitenetter that survey hasn't got verification or report requires. So very handful people have got access about it. No issue of sensitive or blur records.

**Comment by kitenetter** [comment]
Yes, that's why I'm asking about it. What type of data is it collecting, are there any species records? I'm wondering if it will be collecting records that we don't want verifiers to see.

**Comment by DavidRoy** [comment]
It's a survey with National Trust with Michael. No species data, mainly fixed photos

**Comment by kitenetter** [comment]
Thanks. In that case I'll set the sharing agreement to cover reporting + verification + dataflow.

**Comment by kitenetter** [comment]
Can't get my brain in gear today.

I can't set up the sharing agreement until we have the sensitive species workflow in place for website ID 145. I don't think I'm able to adjust that. @johnvanbreda please can you add this website to the ones set up under #1007

**Comment by kitenetter** [comment]
Website agreement added.

We're nearly done I think!

@BirenRathod please can you make the site name attribute mandatory.

**Comment by BirenRathod** [comment]
@kitenetter done.
